contracts/Stabilizer.sol

```diff
@@ -0,0 +1,155 @@
+
+    constructor(ERC20 synth_, ERC20 reserve_, address gov_, uint buyFee_, uint sellFee_, uint supplyCap_) public {
+        require(buyFee_ <= MAX_FEE, "buyFee_ too high");
+        require(sellFee_ <= MAX_FEE, "sellFee_ too high");
+        synth = synth_;
+        reserve = reserve_;
+        governance = gov_;
+        buyFee = buyFee_;
+        sellFee = sellFee_;
+        operator = msg.sender;
+        supplyCap = supplyCap_;
+    }
+
+    modifier onlyOperator {
+        require(msg.sender == operator || msg.sender == governance, "ONLY OPERATOR OR GOV");
+        _;
+    }
+
+    modifier onlyGovernance {
+        require(msg.sender == governance, "ONLY GOV");
+        _;
+    }
+
+    function setOperator(address operator_) public {
+        require(msg.sender == governance || msg.sender == operator, "ONLY GOV OR OPERATOR");
+        operator = operator_;
+    }
+
+    function setBuyFee(uint amount) public onlyOperator {
+        require(amount <= MAX_FEE, "amount too high");
+        buyFee = amount;
+    }
+
+    function setSellFee(uint amount) public onlyOperator {
+        require(amount <= MAX_FEE, "amount too high");
+        sellFee = amount;
+    }
+    
+    function setCap(uint amount) public onlyOperator {
+        supplyCap = amount;
+    }
+
+    function setGovernance(address gov_) public onlyGovernance {
+        governance = gov_;
+    }
+
+    function setStrat(IStrat newStrat) public onlyGovernance {
+        if(address(strat) != address(0)) {
+            uint prevTotalValue = strat.calcTotalValue();
+            strat.divest(prevTotalValue);
+        }
+        reserve.transfer(address(newStrat), reserve.balanceOf(address(this)));
+        newStrat.invest();
+        strat = newStrat;
+    }
+
+    function removeStrat() public onlyGovernance {
+        uint prevTotalValue = strat.calcTotalValue();
+        strat.divest(prevTotalValue);
+
+        strat = IStrat(address(0));
+    }
+
+    function takeProfit() public {
+        uint totalReserves = getTotalReserves();
+        if(totalReserves > supply) {
+            uint profit = totalReserves - supply; // underflow prevented by if condition
+            if(address(strat) != address(0)) {
+                uint bal = reserve.balanceOf(address(this));
+                if(bal < profit) {
+                    strat.divest(profit - bal); // underflow prevented by if condition
+                }
+            }
+            reserve.transfer(governance, profit);
+        }
+    }
+
+    function buy(uint amount) public {
+        if(address(strat) != address(0)) {
+            reserve.transferFrom(msg.sender, address(strat), amount);
+            strat.invest();
+        } else {
+            reserve.transferFrom(msg.sender, address(this), amount);
+        }
+
+        if(buyFee > 0) {
+            uint fee = amount.mul(buyFee).div(MAX_FEE);
+            reserve.transferFrom(msg.sender, governance, fee);
+        }
+
+        synth.mint(msg.sender, amount);
```

Comment:
```suggestion
        synth.mint(msg.sender, amount.sub(fee));
```
Shouldn't we be reducing by the fee here so it doesn't become insolvent?

Reply:
The user gets the Dola `amount` they requested and they pay the extra `fee` in Dai.

Reply:
Hmm, I see your fee changes.

Lets say I buy 10000 DOLA with 10000 USDC with a 1% fee. I get 10000 DOLA, the PSM has 10000 USDC - 100 USD fee = 9900.

Now if I want to trade my 10000 DOLA, I trade into the PSM, but the PSM only has 9900 USDC. If the sell fee is lower than the buy fee, there isn't enough USDC for the trade no?

Reply:
Also, the user does not experience the fee when they buy, but they do when they sell.

Reply:
`Lets say I buy 10000 DOLA with 10000 USDC with a 1% fee. I get 10000 DOLA, the PSM has 10000 USDC - 100 USD fee = 9900.`

The contract gets 10000 USDC then the 1% fee is taken is addition to that amount (see line 119)